independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/update/rewrite/QuarkusUpdatesRepository.java

```diff
@@ -107,6 +107,15 @@ public static FetchResult fetchRecipes(MessageWriter log, MavenArtifactResolver
                         buildTool,
                         propRewritePluginVersion));
 
+                log.debug(String.format(
+                        "Detected dependencies:\n %s ",
+                        recipeDirectoryNames.entrySet().stream()
+                                .map(e -> String.format("%s (%s -> %s)", e.getKey(), e.getValue()[0], e.getValue()[1]))
+                                .sorted().collect(Collectors.joining("\n "))));
+                log.debug(String.format(
+                        "Detected recipe(s):\n %s",
+                        recipes.keySet().stream().sorted().collect(Collectors.joining("\n "))));
```

Comment:
Could you protect this with `isDebugEnabled()`? This would avoid generating the strings when not in debug.

While not in a critical hot path here, it's always better to keep it in mind.